packages/scripts/src/application.ts

```diff
@@ -765,6 +766,10 @@ export class Application {
             : fs.findClosestFileSync(basePath, 'webpack.config.js');
         const baseConfig = (typeof baseConfigPath === 'string' ? require(baseConfigPath) : {}) as webpack.Configuration;
 
+        // We must use a path within the same context of the project for proper module resolutions from the entrypoint
+        const modulesPath = fs.resolve(basePath, 'node_modules');
+        const tmpDirPath = nativeFs.mkdtempSync(fs.join(modulesPath, 'engine-entry'), 'utf8');
```

Comment:
I am just writing the entry points into a "temporary module" which is unique because of mktempSync - and I'm cleaning them up when the compiler is done
I'm not a huge fan of generating module directories on the fly, but I needed a way to guarantee uniqueness  and that is safe to clean up when i'm done with
@AviVahl 

As a side node, maybe we should add that API to file-services?